app/room/page.js

```
@@ -0,0 +1,270 @@
+    setRoomId(generatedRoomId);
+    
+    // Initialize camera and microphone
+    initializeMedia();
+    
+    // Set up signaling and WebRTC
+    setupWebRTC();
+    
+    return () => {
+      // Clean up resources when component unmounts
+      if (localStreamRef.current) {
+        localStreamRef.current.getTracks().forEach(track => track.stop());
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)